src/main/java/org/hibernate/search/develocity/plugins/SurefireConfiguredPlugin.java

```diff
@@ -26,14 +32,56 @@ protected void configureTest(GoalMetadataProvider.Context context) {
 			dependsOnConfigurableJavaExecutable( inputs, context, "jvm", isSkipped( context ),
 					JavaVersions::forJavaExecutable );
 		} );
+
+		configureEnvironmentVariables(context);
+	}
+
+	// Develocity handles environment variables as a big blob by default,
+	// which won't work if some variables point to absolute paths.
+	private void configureEnvironmentVariables(GoalMetadataProvider.Context context) {
+		context.metadata().inputs( inputs -> {
+			// First, override the property to disable handling of environment variables as a blob.
+			// NOTE: ignoring with inputs.ignore( "environmentVariables" ) doesn't work for some reason:
+			// we end up with the goal being marked as "not cacheable"
+			// because "properties were declared both as input and ignored: [environmentVariables]"
+			// NOTE: we get the same result with context.nested( "environmentVariables" ),
+			// which is why we don't use that.
+			inputs.property( MavenMojoExecutionConfig.SUREFIRE_ENVIRONMENT_VARIABLES, "IGNORED" );
+
+			// Then, try to mimic system properties handling.
+			for ( Map.Entry<String, String> envVariable : context.configuration()
+					.getSurefireEnvironmentVariables().entrySet() ) {
+				var key = envVariable.getKey();
+				var keyForDevelocity = MavenMojoExecutionConfig.SUREFIRE_ENVIRONMENT_VARIABLES + "." + key;
+				var value = envVariable.getValue();
+				if ( value == null ) {
+					value = "";
+				}
+				if ( value.startsWith( context.metadata().getSession().getExecutionRootDirectory() ) ) {
+					if ( TEST_INDEXES_PATTERN.matcher( "test-indexes" ).find() ) {
+						// Lucene indexes used in tests -- we don't care about these.
+						inputs.ignore( keyForDevelocity );
```

Comment:
Note this was necessary too, as normalization patterns (see `Normalization.java`, `RuntimeClasspathNormalization`) didn't work. I guess they are unrelated. 

Adding include/excludes didn't work either, and I think that's because those patterns apply to the relative path of elements inside the fileset, rather than to the root of the fileset.